ocaml/xapi/xapi_vgpu_type.ml

```diff
@@ -363,6 +363,33 @@ module Vendor_nvidia = struct
   let pt_when_vgpu = true
   let whitelist_file () = !Xapi_globs.nvidia_whitelist
   let device_id_of_conf conf = conf.identifier.Identifier.pdev_id
+  
+  let get_host_driver_version () = try Scanf.sscanf (Unix.readlink !Xapi_globs.nvidia_host_driver_file) "libnvidia-vgpu.so.%s" (fun x -> x)
+    with _ -> info "Can not get the file version of %s, use the default value" !Xapi_globs.nvidia_host_driver_file; Xapi_globs.nvidia_default_host_driver_version
+
+  let is_host_driver_support_multi_vgpu ~host_driver_version ~supported_versions =
```

Comment:
This function is so complicated that it is impossible to review... :( Does it have to have all these cases with `+` for major and minor separately? Without that, and just a + for the whole thing, you could just do something like this?
```
let is_host_driver_support_multi_vgpu ~host_driver_version ~supported_versions =
  let version = float_of_string host_driver_version in
    List.fold_left (fun result pattern ->
      match List.index_opt pattern '+' with
      | Some i -> result || (float_of_string (String.sub pattern i) <= version)
      | None   -> result || (float_of_string pattern = version)
    ) false supported_versions
```

Reply:
I tend to agree. Do we really need the flexibility the mechanism implements? Is there a reason to believe that any higher version than a minimum is not usable? I can imagine a number of simpler approaches:

* Have a minimum version and accept all versions greater than that. We can always add a blacklist of versions that we would want to exclude.
* Use a list of regular expressions. Any version that matches at least one of the regular expressions is accepted.

Reply:
```

Reply:
As I know, `430.27` does not support multiple vGPU.

Reply:
It would be easy to exclude `430.27` by creating a match to `false` as the 1st or 2nd entry. That is the point of this: it's easy because it uses to full power of OCaml. The disadvantage is that it cannot be controlled from the command line - but even that would not be difficult.

Reply:
Thanks Lindig for your advice.  This is still under discussing, but Rob's suggestion looks good so far. 
let's wait for the final decision, and perhaps we can just take Rob's advice if possible.

Reply:
Thanks @robhoes  and @lindig  for the discussion about this PR.
As @robhoes suggestion looks good so far so good, and the logic of how to determine whether a host  driver is encapsulated into a single function (which is easy to change if necessary), can I suggest we just take Rob's suggestion in this PR and continute the review. 

I changed the `List.fold_left` to `List.exist` as we does not need to iterate all the elements if there is one elements return true.